File: backend/healthmonitor/src/main/java/com/devgirls/healthmonitor/dto/HealthDataDTO.java
package com.devgirls.healthmonitor.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class HealthDataDTO {
    private Long userId;
    private LocalDateTime timestamp;
    private Integer heartRate;
    private Integer steps;
    private Double calories;
    private Double sleepHours;
    private String source;
}

File: ios/HealthSystem/HealthSystem/HealthDataDTO.swift
//
//  HealthDataDTO.swift
//  HealthSystem
//
//  Created by Aruuke Turgunbaeva on 22/10/25.
//

import Foundation

struct HealthDataDTO: Codable {
    let userId: Int
    let timestamp: String       // ISO8601 string
    let heartRate: Int?
    let steps: Int?
    let calories: Double?
    let sleepHours: Double?
    let source: String?
}
